vhost-device-vsock/src/vhu_vsock_thread.rs

```diff
@@ -448,8 +448,8 @@ impl VhostUserVsockThread {
             .read_exact(&mut buf[..MIN_READ_LEN])
             .map_err(Error::UnixRead)?;
 
-        let mut read_len = MIN_READ_LEN;
-        while buf[read_len - 1] != b'\n' && read_len < buf.len() {
+        let mut read_len = 0;
```

Comment:
I don't think this works, because this way we are going to overwrite the buffer we read above, maybe we need to delete the previous line. Or find a better way.

Reply:
is it okay if we just always read 1 byte at a time, instead of the inital 10 bytes that we read. Since it's only used for events i don't see much of performance implications too

Reply:
yep, I think it is fine.
Otherwise we can use some `std::io` module for buffered read. Honestly, I don't remember why we implemented in that way.

Reply:
i am doing something like this 
```rust 
let mut buf = Vec::with_capacity(MAX_BUF as usize);
let mut reader = BufReader::new(stream).take(MAX_BUF);

let n = reader
    .read_until(b'\n', &mut buf)
    .map_err(Error::UnixRead)?;
```
I know it does a heap allocation instead of doing it on stack like we were previously doing because `read_until` takes a &mut Vec<u8>. Is that an issue?

Reply:
Personally I'd approach it as follows:

1. read and discard the entire byte sequence `b"CONNECT " == [b'C', b'O', b'N', b'N', b'E', b'C', b'T', b' ']` with `ignore_ascii_case` if it's not necessary for `CONNECT` to be uppercase. Bail out early if something doesn't match. 
2. read and save bytes to stack allocated `buf` until '`b'\n'` is received or some invalid state. Handle errors if any.
3. convert byte `buf` to u32 directly from `&[u8]`: check if all bytes `u8::is_ascii_digit` return `true`, then use `unsafe ` and `std::str::from_utf8_unchecked` since a valid array of ASCII characters is valid UTF8, so the conversion is safe.

Reply:
I don't have a particular preference, maybe even I would have done the same as @epilys suggested. But since it's not a critical point, I'd say let's do whatever comes out easiest.

Reply:
++ Most definitely, if it works then it's fine even if it does heap allocations, no need to micro optimize everything :)